Fedora/hpx.spec

```diff
@@ -165,9 +208,17 @@ HPX compiled with Open MPI, package incl. examples
 for mpi in '' openmpi mpich ; do
   test -n "${mpi}" && module load mpi/${mpi}-%{_arch}
   mkdir -p ${mpi:-serial}
-  pushd ${mpi:-serial}
+  pushd ${s:-.}/${mpi:-serial}
+  if [ -n "$mpi" ]; then
+  export CC=mpicc
+  export CXX=mpicxx
+  %{cmake} -DHPX_WITH_PARCELPORT_MPI=0N -DLIB=${MPI_LIB} %{?cmake_opts:%{cmake_opts}} ..
+  else
+  export CC=gcc
+  export CXX=g++
```

Comment:
Do we really need to export this? Wasn't cmake picking up the right compilers?

Reply:
Yes, I had to export them to avoid linker errors for libhpx. At least after exporting them the linking error vanished.

Reply:
Is it important to remove them or just cosmetic?

Reply:
I did a [build](https://kojipkgs.fedoraproject.org//work/tasks/9305/30769305/build.log) without the export resulting

```
/usr/bin/ld: CMakeFiles/hpx.dir/version.cpp.o: in function `MPI::Op::Init(void (*)(void const*, void*, int, MPI::Datatype const&), bool)':
/usr/include/openmpi-i386/openmpi/ompi/mpi/cxx/op_inln.h:121: undefined reference to `ompi_mpi_cxx_op_intercept'
/usr/bin/ld: CMakeFiles/hpx.dir/version.cpp.o: in function `MPI::Intracomm::Clone() const':
/usr/include/openmpi-i386/openmpi/ompi/mpi/cxx/intracomm_inln.h:23: undefined reference to `MPI::Comm::Comm()'
/usr/bin/ld: CMakeFiles/hpx.dir/version.cpp.o: in function `MPI::Graphcomm::Clone() const':
/usr/include/openmpi-i386/openmpi/ompi/mpi/cxx/intracomm.h:25: undefined reference to `MPI::Comm::Comm()'
/usr/bin/ld: CMakeFiles/hpx.dir/version.cpp.o: in function `MPI::Cartcomm::Sub(bool const*) const':
/usr/include/openmpi-i386/openmpi/ompi/mpi/cxx/intracomm.h:25: undefined reference to `MPI::Comm::Comm()'
/usr/bin/ld: CMakeFiles/hpx.dir/version.cpp.o: in function `MPI::Intracomm::Create_graph(int, int const*, int const*, bool) const':
/usr/include/openmpi-i386/openmpi/ompi/mpi/cxx/intracomm.h:25: undefined reference to `MPI::Comm::Comm()'
/usr/bin/ld: CMakeFiles/hpx.dir/version.cpp.o: in function `MPI::Cartcomm::Clone() const':
/usr/include/openmpi-i386/openmpi/ompi/mpi/cxx/intracomm.h:25: undefined reference to `MPI::Comm::Comm()'
/usr/bin/ld: CMakeFiles/hpx.dir/version.cpp.o:/usr/include/openmpi-i386/openmpi/ompi/mpi/cxx/intracomm.h:25: more undefined references to `MPI::Comm::Comm()' follow
/usr/bin/ld: CMakeFiles/hpx.dir/version.cpp.o:(.data.rel.ro._ZTVN3MPI8DatatypeE[_ZTVN3MPI8DatatypeE]+0x3c): undefined reference to `MPI::Datatype::Free()'
/usr/bin/ld: CMakeFiles/hpx.dir/version.cpp.o:(.data.rel.ro._ZTVN3MPI3WinE[_ZTVN3MPI3WinE]+0x24): undefined reference to `MPI::Win::Free()'
```
and thus I will add them again.

Reply:
I think that points towards a bug in hpx's build system. The above error means there is is a
```cmake
target_link_libraries(hpx ${MPI_LIBRARIES})
```
missing.

Reply:
Strange, it has it here: https://github.com/STEllAR-GROUP/hpx/blob/master/plugins/parcelport/mpi/CMakeLists.txt#L88

Releated question, `-DHPX_WITH_PARCELPORT_TCP=0N` for the non-mpi version, or is that the default?

Reply:
Only been following along casually so I might be way off, but https://github.com/STEllAR-GROUP/hpx/issues/3475 may be related. `-DHPX_WITH_PARCELPORT_MPI=0N` is also definitely a typo (zero instead of the letter O), but not sure if that will make a difference.

Reply:
I started a rebuild with the `ON`, since I found some questions related to the issues and others did the export for their rpm packages too, I think we can keep it.

Reply:
Yeah no big deal, we can always remove it later, but I think @msimberg i right it is STEllAR-GROUP/hpx#3475!